File: sadik-backend/app/schemas/feedback.py
from pydantic import BaseModel, Field
from typing import Optional


class FeedbackSubmit(BaseModel):
    type: str = Field(..., pattern="^(bug|feature|other)$")
    body: str = Field(..., min_length=10, max_length=2000)
    screenshot_base64: Optional[str] = None
    app_version: Optional[str] = None
    os_info: Optional[str] = None
    current_page: Optional[str] = None


class FeedbackResponse(BaseModel):
    id: int
    ok: bool

File: sadik-app/src/api/feedback.ts
import { http } from './http';

export interface FeedbackBody {
  type: 'bug' | 'feature' | 'other';
  body: string;
  screenshot_base64?: string;
  app_version?: string;
  os_info?: string;
  current_page?: string;
}

export const feedbackApi = {
  submit: (data: FeedbackBody) =>
    http.post<{ id: number; ok: boolean }>('/api/feedback', data).then((r) => r.data),
};
